packages/components/src/font-size-picker/index.tsx

```diff
@@ -38,6 +42,49 @@ import { T_SHIRT_NAMES } from './constants';
 
 const DEFAULT_UNITS = [ 'px', 'em', 'rem', 'vw', 'vh' ];
 
+const shouldUseSelectOverToggle = ( howManyfontSizes: number ) =>
+	howManyfontSizes > 5;
+
+const getPickerType = (
+	showCustomValuePicker: boolean,
+	fontSizes: FontSize[]
+): FontSizePickerType => {
+	if ( showCustomValuePicker ) {
+		return 'custom';
+	}
+
+	return shouldUseSelectOverToggle( fontSizes.length )
+		? 'select'
+		: 'togglegroup';
+};
+
+const getHeaderHint = (
+	currentPickerType: FontSizePickerType,
+	selectedFontSize: FontSize | undefined,
+	fontSizes: FontSize[]
+) => {
+	if ( currentPickerType === 'custom' ) {
+		return __( 'Custom' );
+	}
+
+	if ( ! shouldUseSelectOverToggle( fontSizes.length ) ) {
+		if ( selectedFontSize ) {
+			return (
+				selectedFontSize.name ||
+				T_SHIRT_NAMES[ fontSizes.indexOf( selectedFontSize ) ]
+			);
+		}
+		return '';
```

Comment:
I know this is pre-existing, but I wasn't sure why in this case we don't display the common unit (like we do below) and if this is intentional.

Reply:
I'm not sure if this was intentional or not — I would probably need to look in the git history and see if this aspect was discussed actively in previous PRs. Probably out of scope for this refactor, though — for now, I've decided to keep the `''` fallback around to preserve the same behaviour as on `trunk`.